### PR TITLE
ocamlPackages.pbkdf: 1.1.0 -> 1.2.0

### PR DESCRIPTION
**Pull request by ulrikstrid**
###### Description of changes

Updates pbkdf

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/ocaml-modules/pbkdf/default.nix [hunk 1]
```diff
@@ -12,12 +12,13 @@ buildDunePackage rec {
 
   src = fetchFromGitHub {
     owner = "abeaumont";
-    repo = "ocaml-pbkdf";
+    repo = "ocaml-${pname}";
     rev = version;
     sha256 = "sha256-dGi4Vw+7VBpK/NpJ6zdpogm+E6G/oJovXCksJBSmqjI=";
   };
```
Comment (superherointj) [inline pkgs/development/ocaml-modules/pbkdf/default.nix:19]:
```suggestion
    minimalOCamlVersion = "4.07";
```
https://github.com/ocaml/opam-repository/blob/master/packages/pbkdf/pbkdf.1.2.0/opam#L15

File: pkgs/development/ocaml-modules/pbkdf/default.nix [hunk 2]
```diff
@@ -12,12 +12,13 @@ buildDunePackage rec {
 
   src = fetchFromGitHub {
```
Comment (superherointj) [inline pkgs/development/ocaml-modules/pbkdf/default.nix:13]:
```suggestion
  src = fetchFromGitHub {
```
Replace with:
```
  src = fetchzip {
    url = "https://github.com/abeaumont/ocaml-pbkdf/releases/download/1.1.0/pbkdf-1.1.0.tbz";
    sha256 = "sha256-2WdCwqXEE9hMPk71j0tNa2j6jsn++Z8yU7g1GBOlPmA=";
  };
```

File: pkgs/development/ocaml-modules/pbkdf/default.nix [hunk 3]
```diff
@@ -12,12 +12,13 @@ buildDunePackage rec {
...
     owner = "abeaumont";
-    repo = "ocaml-pbkdf";
+    repo = "ocaml-${pname}";
     rev = version;
     sha256 = "sha256-dGi4Vw+7VBpK/NpJ6zdpogm+E6G/oJovXCksJBSmqjI=";
   };
```
Comment (superherointj) [inline pkgs/development/ocaml-modules/pbkdf/default.nix:18]:
```suggestion
  src = fetchzip {
    url = "https://github.com/abeaumont/ocaml-pbkdf/releases/download/${version}/pbkdf-${version}.tbz";
    sha256 = "sha256-2WdCwqXEE9hMPk71j0tNa2j6jsn++Z8yU7g1GBOlPmA=";
  };
```
To fix version issue in META file.

File: pkgs/development/ocaml-modules/pbkdf/default.nix [hunk 4]
```diff
@@ -12,12 +12,13 @@ buildDunePackage rec {
...
 
-  propagatedBuildInputs = [ cstruct mirage-crypto ];
+  propagatedBuildInputs = [ mirage-crypto ];
+  buildInputs = [ cstruct ];
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/pbkdf/default.nix:21]: This change looks wrong (and the commit message also).

File: pkgs/development/ocaml-modules/pbkdf/default.nix [hunk 5]
```diff
@@ -10,13 +10,12 @@ buildDunePackage rec {
   pname = "pbkdf";
   version = "1.2.0";
 
-  src = fetchFromGitHub {
-    owner = "abeaumont";
-    repo = "ocaml-pbkdf";
-    rev = version;
+  src = fetchzip {
+    url = "https://github.com/abeaumont/ocaml-pbkdf/archive/${version}.tar.gz";
```
Comment (superherointj) [inline pkgs/development/ocaml-modules/pbkdf/default.nix:14]:
```suggestion
    url = "https://github.com/abeaumont/ocaml-pbkdf/archive/${version}.zip";
```

Comment (superherointj) [reply pkgs/development/ocaml-modules/pbkdf/default.nix:14]: (I'm attempting to fix version shown in META with this change.)

**Review by superherointj — APPROVED** [review]

**Comment by superherointj** [comment]
Just noticed one thing:
It was not possible to solve META version here because we are getting the sources from a tag and not from a release, which is unavailable for 1.2.0. It should have been: [pbkdf-1.1.0.tbz](https://github.com/abeaumont/ocaml-pbkdf/releases/download/1.1.0/pbkdf-1.1.0.tbz)
Notice the `.tbz`.
Anyway, I guess there is nothing more to do then.

File: pkgs/development/ocaml-modules/pbkdf/default.nix (hunk 3 above)
Comment (superherointj) [reply pkgs/development/ocaml-modules/pbkdf/default.nix]: On my understanding: `.tbz` files have metadata that is used in OCaml for version (check package's META file to understand). Ideally we should be fetching `.tbz` to retrieve the version metadata.
@vbgl Thoughts on @SuperSandro2000 feedback?

**Comment by ulrikstrid** [comment]
@superherointj the problem is that there's no release for 1.2.0, only a tag: https://github.com/abeaumont/ocaml-pbkdf/releases

**Pull request merged**